contracts/ERC721Collateralizer.sol

```diff
@@ -0,0 +1,372 @@
+
+        // Ensure a valid form of collateral is tied to this agreement id.
+        require(collateralTokenAddress != address(0));
+
+        // Seizure can only occur if the collateral has yet to be withdrawn.
+        // When we withdraw collateral, we reset the collateral agreement
+        // in a gas-efficient manner by resetting the address of the collateralizer to 0.
+        require(agreementToCollateralizer[agreementId] != address(0));
+
+        // Ensure debt is in a state of default.
+        require(
+            termsContract.getExpectedRepaymentValue(
+                agreementId,
+                block.timestamp
+            ) > termsContract.getValueRepaidToDate(agreementId)
+        );
+
+        // Mark agreement's collateral as withdrawn by setting the agreement's
+        // collateralizer to 0x0.
+        delete agreementToCollateralizer[agreementId];
+
+        // Determine beneficiary of the collateral to be seized.
+        address beneficiary = debtRegistry.getBeneficiary(agreementId);
+
+        ERC721 collateralToken = ERC721(collateralTokenAddress);
+
+        // Transfer the collateral this contract was holding in escrow to beneficiary.
+        collateralToken.transferFrom(address(this), beneficiary, collateralTokenID);
+
+        // Log the seizure event.
+        CollateralSeized(
+            agreementId,
+            beneficiary,
+            collateralTokenAddress,
+            collateralTokenID
+        );
+    }
+
+    /**
+     * Adds an address to the list of agents authorized
+     * to invoke the `collateralize` function.
+     */
+    function addAuthorizedCollateralizeAgent(address agent)
+        public
+        onlyOwner
+    {
+        collateralizationPermissions.authorize(agent, CONTEXT);
+    }
+
+    /**
+     * Removes an address from the list of agents authorized
+     * to invoke the `collateralize` function.
+     */
+    function revokeCollateralizeAuthorization(address agent)
+        public
+        onlyOwner
+    {
+        collateralizationPermissions.revokeAuthorization(agent, CONTEXT);
+    }
+
+    /**
+    * Returns the list of agents authorized to invoke the 'collateralize' function.
+    */
+    function getAuthorizedCollateralizeAgents()
+        public
+        view
+        returns (address[])
+    {
+        return collateralizationPermissions.getAuthorizedAgents();
+    }
+
+    /**
+     * Unpacks collateralization-specific parameters from their tightly-packed
+     * representation in a terms contract parameter string.
+     */
+    function unpackCollateralParametersFromBytes(bytes32 parameters)
+        public
+        pure
+        returns (uint, uint)
+    {
+        bytes32 collateralTokenIndexShifted =
+            parameters & 0x0000000000000000000000000000000000000fffffffffffff00000000000000;
+        bytes32 tokenIdShifted =
+            parameters & 0x00000000000000000000000000000000000000000000000000ffffffffffffff;
```

Comment:
Some more comments here a la what we have in the SimpleInterestTermsContract unpacking function would be very useful.  Doing a lot of digit counting over here :)